orchestration/dagster_orchestration/hca_manage/manage.py

```diff
@@ -40,18 +40,23 @@ def __init__(self, environment: str, data_repo_client: RepositoryApi, project: O
                      "dev": "https://jade.datarepo-dev.broadinstitute.org"}
         self.base_url = jade_urls[environment]
 
-        # use application default credentials to seamlessly work across monster devs
-        # assumes `gcloud auth application-default login` has been run
-        creds, _ = google.auth.default()
-        self.gcp_creds = creds
-
         self.reader_list = {
             "dev": ["hca-snapshot-readers@dev.test.firecloud.org"],
             "prod": ["hca-snapshot-readers@firecloud.org"]
         }[environment]
 
+        self._gcp_creds = None
+
+    def gcp_creds(self):
+        # use application default credentials to seamlessly work across monster devs
+        # assumes `gcloud auth application-default login` has been run
+        if not self._gcp_creds:
```

Comment:
❓ Why shift to lazy-loading the app creds?

Reply:
eager loading them causes authentication errors in the tests on initialization, even if we've mocked out all external api calls, whereas lazy loading doesn't try to call out to gcloud unless we are trying to make a live gcloud request